src/Bicep.Core/TypeSystem/FunctionFlags.cs

```diff
@@ -38,13 +38,18 @@ public enum FunctionFlags
         /// <summary>
         /// The function can be used as a resource decorator.
         /// </summary>
-        ResoureDecorator = 1 << 4,
+        ResourceDecorator = 1 << 4,
 
         /// <summary>
         /// The function can be used as a module decorator.
         /// </summary>
         ModuleDecorator = 1 << 5,
 
+        /// <summary>
+        /// The function can be used a resource or module decorator.
+        /// </summary>
+        ResourceOrModuleDecorator = ResourceDecorator | ModuleDecorator,
```

Comment:
>ResourceOrModuleDecorator [](start = 8, length = 25)

The TypeAssignmentVisitor changes explain why I explicitly declared the combination.